### PR TITLE
Catch errors in `TaskListenerDecorator.decorate` even when wrapped with `merge`

### PR DESCRIPTION
**Pull request by jglick**
See https://github.com/jenkinsci/gradle-plugin/pull/318 for context. Do not risk having a decorator be completely skipped just because an unrelated decorator was erroneous.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/log/TaskListenerDecorator.java [hunk 1]
```diff
@@ -219,6 +219,17 @@ private static class ConsoleLogFilterAdapter extends TaskListenerDecorator {
 
     }
 
+    private static OutputStream decorateAll(OutputStream base, List<TaskListenerDecorator> decorators) {
+        for (TaskListenerDecorator decorator : decorators) {
+            try {
+                base = decorator.decorate(base);
+            } catch (Throwable x) {
+                LOGGER.log(Level.WARNING, null, x);
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/workflow/log/TaskListenerDecorator.java:227]: Whilst this defends against the issue I am concerned this error will never be picked up - most people are not looking at jenkins logs.  Without a build to log at least prompt the build user to ask the admin to look at the logs.

Having said that - I have no idea how we would get the charset to print something here - without the charset we risk corrupting the entire output :(

**Review by jtnord — APPROVED** [review]

**Review by rsandell — APPROVED** [review]

File: src/main/java/org/jenkinsci/plugins/workflow/log/TaskListenerDecorator.java (hunk 1 above)
Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/workflow/log/TaskListenerDecorator.java:229]: ⇒ d191982eb38dd0dc85a75c2378e17b3a278e1ad8

> the charset to print something here

⇒ 696e4f8900771738d075bbc38524ff0de7899a77

File: src/main/java/org/jenkinsci/plugins/workflow/log/TaskListenerDecorator.java [hunk 2]
```diff
@@ -219,6 +221,18 @@ private static class ConsoleLogFilterAdapter extends TaskListenerDecorator {
...
+                Functions.printStackTrace(x, new PrintStream(base, true, StandardCharsets.UTF_8));
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/workflow/log/TaskListenerDecorator.java:230]: is the pipeline console log always `UTF-8` and never platform dependant?

**Review by jtnord — APPROVED** [review]

**Review by jglick — COMMENTED** [review]
Got GitHub into a weird corner case where I have zero pending changes yet still have a pending review I need to cancel and cannot do so without leaving a comment.

**Pull request merged**